src/components/WarningModal.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * This file is part of OrangeHRM
+ *
+ * Copyright (C) 2020 onwards OrangeHRM (https://www.orangehrm.com/)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import React from 'react';
+import Modal, {ModalProps} from 'react-native-modal';
+import {View, StyleSheet, TouchableOpacity, SafeAreaView} from 'react-native';
+import {Icon} from 'native-base';
+import Text from './DefaultText';
+import {setItem} from 'store/storage/actions';
+import {RootState} from 'store';
+import {selectMyInfo} from 'store/auth/selectors';
+import {connect} from 'react-redux';
+import {WARNING_MODAL_STATUS} from 'services/storage';
+
+const WarningModal = (props: React.PropsWithChildren<WarningModalProps>) => {
```

Comment:
not used theme parameters, and reusable components. but accepting as this component will delete once mobile app compatible with OrangeHRM 5.x